src/transaction/check-cash.ts

```diff
@@ -0,0 +1,50 @@
+import * as utils from './utils'
+const ValidationError = utils.common.errors.ValidationError
+const toRippledAmount = utils.common.toRippledAmount
+import {validate} from '../common'
+import {Instructions, Prepare} from './types'
+import {Amount} from '../common/types/objects'
+
+type CheckCash = {
+  checkID: string,
+  amount?: Amount,
+  deliverMin?: Amount
+}
+
+function createCheckCashTransaction(account: string,
+  checkCash: CheckCash
+): Object {
+  const txJSON: any = {
+    Account: account,
+    TransactionType: 'CheckCash',
+    CheckID: checkCash.checkID
+  }
+
+  if (checkCash.amount && checkCash.deliverMin) {
+    throw new ValidationError('"amount" and "deliverMin" properties on '
```

Comment:
We tend to do data validation as the first thing in the function. It's a good general rule so that you don't have to worry about what logic is safe to do pre-validation vs. post-validation (ex: if someone later had to add something to the txJSON definition above).